android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -100,6 +101,11 @@ public CodePush(String deploymentKey, Activity mainActivity, boolean isDebugMode
         }
 
         initializeUpdateAfterRestart();
+        if (currentInstance != null) {
+            CodePushUtils.log("More than one CodePush instances has been initialized, please use the non-static method codePush.getBundleUrlInternal() to get the correct bundleURL for a particular instance.");
```

Comment:
We know of one user eg. https://github.com/Microsoft/react-native-code-push/pull/255 that is thinking of having multiple ReactActivity in their app. While we don't currently support this scenario now, I wanted to keep the architecture flexible for enabling that in the future.